src/riot-rs-threads/src/arch/cortex_m.rs

```diff
@@ -1,78 +1,76 @@
+use super::Arch;
 use core::arch::asm;
 use core::ptr::write_volatile;
 use cortex_m::peripheral::SCB;
 use critical_section::CriticalSection;
 
 use crate::{cleanup, THREADS};
 
-/// Sets up the stack for newly created threads and returns the sp.
-///
-/// After running this, the stack should look as if the thread was
-/// interrupted by an ISR.
-/// The exact order in which Cortex-M pushes the registers to the stack when
-/// entering the ISR is:
-///
-/// +---------+ <- sp
-/// |   r0    |
-/// |   r1    |
-/// |   r2    |
-/// |   r3    |
-/// |   r12   |
-/// |   LR    |
-/// |   PC    |
-/// |   PSR   |
-/// +---------+
-///
-/// This function sets up the stack so when the context is switched to this thread,
-/// it starts executing `func` with argument `arg`.
-/// Furthermore, it sets up the link-register with the [`crate::cleanup`] function that
-/// will be executed after the thread function returned.
-pub(crate) fn setup_stack(stack: &mut [u8], func: usize, arg: usize) -> usize {
-    let stack_start = stack.as_ptr() as usize;
+pub struct Cpu;
```

Comment:
bikeshedding: `Cpu` ... does anyone have another idea how to name this?

Reply:
Agreed that CPU isn't ideal. I had `Arch` initially, but that conflicted with the trait name...

Reply:
alright, let's sit this one out. :)

Reply:
> Is this per-core or per CPU? If this is per-core, then it could be (Cpu|Processor)Core (not very familiar with that part of the codebase yet, this may not make sense).

It is just an empty struct that the Arch trait is implemented on.